cabal-install/cabal-install.cabal

```diff
@@ -189,6 +189,11 @@ Flag network-uri
   description:  Get Network.URI from the network-uri package
   default:      True
 
+Flag debug-assertions
```

Comment:
Or, given that we already have `debug-tracetree`, we could also have `debug-assertions` for `ghc-options: -fno-ignore-asserts` and `debug-expensive-assertions` for expensive ones.